src/doom/d_main.c

```diff
@@ -1098,25 +1117,109 @@ static void InitGameVersion(void)
         }
     }
 
-    // Deathmatch 2.0 did not exist until Doom v1.4
-    if (gameversion <= exe_doom_1_2 && deathmatch == 2)
-    {
-        deathmatch = 1;
-    }
-    
-    // The original exe does not support retail - 4th episode not supported
+    // Version-specific differences
+    // Each entry is the last version where the statement is true.
+    switch (gameversion)
+    {
+        case exe_doom_1_0:
+            // Different trig tables
+            ChangeTrigTablesToDoom1_0();
+
+            // Skull keys still spawn in deathmatch
+            mobjinfo[MT_MISC7].flags &= ~MF_NOTDMATCH;
+            mobjinfo[MT_MISC8].flags &= ~MF_NOTDMATCH;
+            mobjinfo[MT_MISC9].flags &= ~MF_NOTDMATCH;
+
```

Comment:
Where a `switch() {...}` fallthrough is intentional, please add a comment that says something like `// fallthrough` - otherwise this looks like a mistake.

Reply:
Moved to `d_compat.c` and changed in https://github.com/chocolate-doom/chocolate-doom/pull/1207/commits/d8f6bcf685da5f096480d21cc0162de06d1b6f24 .